### PR TITLE
Feat: add auth with oidc

### PR DESCRIPTION
**Pull request by laurentC35**

### CODE REVIEW

File: src/main/java/fr/insee/publicenemy/api/application/web/auth/AuthenticationUserHelper.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class AuthenticationUserHelper implements AuthenticationHelper {
+    @Override
+    public String getUserToken() {
+        if(getAuthenticationPrincipal() instanceof JwtAuthenticationToken auth) {
+            return auth.getToken().getTokenValue();
+        }
+        throw new AuthenticationTokenException("Cannot retrieve token for the user. Ensure you have not disabled oidc with pilotage api enabled");
```
Comment (davdarras) [inline src/main/java/fr/insee/publicenemy/api/application/web/auth/AuthenticationUserHelper.java:19]: change message here maybe =)

File: src/main/java/fr/insee/publicenemy/api/configuration/SwaggerBeanConfig.java [hunk 2]
```diff
@@ -1,22 +0,0 @@
...
-
-@Configuration
-public class SwaggerBeanConfig {
-    
-    /**
-     * Used to workaround the issue where
-     * <a href="https://github.com/swagger-api/swagger-ui/issues/6462">swagger does not handle correctly multipart request</a>
-     * @param converter Jackson converter
-     */  
-    public SwaggerBeanConfig(MappingJackson2HttpMessageConverter converter) {
```
Comment (davdarras) [inline src/main/java/fr/insee/publicenemy/api/configuration/SwaggerBeanConfig.java:17]: not sure about this one. there is a workaround here about multipart request. is it resolved ?

File: src/main/java/fr/insee/publicenemy/api/configuration/auth/NoAuthSecurityConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                .build();
+    }
+
+    @Bean
+    @Order(1)
+    protected SecurityFilterChain filterPublicUrlsChain(HttpSecurity http, ApplicationProperties applicationProperties) throws Exception {
+        return publicSecurityFilterChainConfiguration.buildSecurityPublicFilterChain(http, applicationProperties.publicUrls());
+    }
+
+    @Bean
```
Comment (davdarras) [inline src/main/java/fr/insee/publicenemy/api/configuration/auth/NoAuthSecurityConfiguration.java:62]: remove this as you're not using resttemplate. need to add the same mechanism for webclient though

File: src/main/java/fr/insee/publicenemy/api/PublicEnemyApplication.java [hunk 4]
```diff
@@ -4,8 +4,12 @@
 import org.springframework.boot.builder.SpringApplicationBuilder;
 
 import fr.insee.publicenemy.api.configuration.PropertiesLogger;
+import org.springframework.boot.context.properties.ConfigurationPropertiesScan;
+import org.springframework.transaction.annotation.EnableTransactionManagement;
 
-@SpringBootApplication
+@SpringBootApplication(scanBasePackages = "fr.insee.publicenemy")
+@EnableTransactionManagement
```
Comment (davdarras) [inline src/main/java/fr/insee/publicenemy/api/PublicEnemyApplication.java:11]: outch didn't see this one

File: src/main/java/fr/insee/publicenemy/api/configuration/auth/OidcSecurityConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+        jwtAuthenticationConverter.setJwtGrantedAuthoritiesConverter(jwtGrantedAuthoritiesConverter(oidcProperties, roleProperties));
+        return jwtAuthenticationConverter;
+    }
+
+    Converter<Jwt, Collection<GrantedAuthority>> jwtGrantedAuthoritiesConverter(OidcProperties oidcProperties, RoleProperties roleProperties) {
+        return new GrantedAuthorityConverter(oidcProperties, roleProperties);
+    }
+
+    @Bean
+    protected RestTemplate restTemplateQueen(AuthenticationHelper authenticationHelper) {
```
Comment (davdarras) [inline src/main/java/fr/insee/publicenemy/api/configuration/auth/OidcSecurityConfiguration.java:97]: same here

File: src/main/java/fr/insee/publicenemy/api/configuration/rest/RestTemplateAddJsonHeaderInterceptor.java [hunk 6]
```diff
@@ -0,0 +1,22 @@
+package fr.insee.publicenemy.api.configuration.rest;
```
Comment (davdarras) [inline src/main/java/fr/insee/publicenemy/api/configuration/rest/RestTemplateAddJsonHeaderInterceptor.java:1]: class not needed,

File: src/main/java/fr/insee/publicenemy/api/configuration/rest/RestTemplateTokenInterceptor.java [hunk 7]
```diff
@@ -0,0 +1,28 @@
+package fr.insee.publicenemy.api.configuration.rest;
```
Comment (davdarras) [inline src/main/java/fr/insee/publicenemy/api/configuration/rest/RestTemplateTokenInterceptor.java:1]: change this to add an "interceptor like class" for webclient

File: src/main/java/fr/insee/publicenemy/api/configuration/swagger/role/DisplayRolesOnSwaggerUI.java [hunk 8]
```diff
@@ -0,0 +1,43 @@
+package fr.insee.publicenemy.api.configuration.swagger.role;
```
Comment (davdarras) [inline src/main/java/fr/insee/publicenemy/api/configuration/swagger/role/DisplayRolesOnSwaggerUI.java:1]: maybe you don't need this. It is used to display roles allowed to use endpoints in swagger UI

File: src/main/java/fr/insee/publicenemy/api/configuration/swagger/role/RoleUIMapper.java [hunk 9]
```diff
@@ -0,0 +1,19 @@
+package fr.insee.publicenemy.api.configuration.swagger.role;
```
Comment (davdarras) [inline src/main/java/fr/insee/publicenemy/api/configuration/swagger/role/RoleUIMapper.java:1]: same here, used for swagger ui

File: src/main/java/fr/insee/publicenemy/api/configuration/auth/AuthorityRole.java [hunk 10]
```diff
@@ -0,0 +1,11 @@
+package fr.insee.publicenemy.api.configuration.auth;
```
Comment (davdarras) [inline src/main/java/fr/insee/publicenemy/api/configuration/auth/AuthorityRole.java:1]: maybe not needed. You can just use hasRole on @PreAuthorize annotations. If you don't have a lot of endpoints, juste configure them in OidcSecurityConfiguration

File: src/main/resources/application.yaml [hunk 11]
```diff
@@ -57,8 +84,15 @@ application:
...
-    allowed-origins:
+  public-urls:
+    - /swagger-ui/**
+    - /v3/api-docs/**
+    - /api/healthcheck
+    - /
+  roles:
+    admin: 'admin'
+    designer: 'designer'
+  corsOrigins: '*'
```
Comment (davdarras) [inline src/main/resources/application.yaml:95]: corsOrigins should be empty here. There's a check on this property when loading app to force setting this property

File: src/main/java/fr/insee/publicenemy/api/configuration/SwaggerBeanConfig.java (hunk 2 above)
Comment (laurentC35) [reply src/main/java/fr/insee/publicenemy/api/configuration/SwaggerBeanConfig.java:17]: I'm not sure it's resolved, buy i move this configuration ton SpringDocConfiguration

**Pull request merged**